### PR TITLE
fix(eslint-plugin): [no-require-imports] report only global `require`

### LINKED ISSUE #3713: [no-require-imports] `require` created by `createRequire` should be ignored
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-require-imports": 2
  }
}
```

```TS
import { createRequire } from 'module'

const require = createRequire()

require('remark-preset-prettier')
```

**Expected Result**

No error

**Actual Result**

```log
error  A `require()` style import is forbidden   @typescript-eslint/no-require-imports
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.0` |
| `@typescript-eslint/parser`        | `4.29.0` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `12.22.1` |

### PR DESCRIPTION
**Pull request by rafaelss95**
Fixes #3713.

### CODE REVIEW

File: packages/eslint-plugin/src/rules/no-require-imports.ts [hunk 1]
```diff
@@ -35,3 +39,28 @@ export default util.createRule({
...
   },
 });
+
+/**
+ * Finds the variable by a given name in a given scope and its upper scopes.
+ * @param initScope A scope to start find.
+ * @param name A variable name to find.
+ * @returns A found variable or `null`.
+ */
+function getVariableByName(
```
Comment (rafaelss95) [inline packages/eslint-plugin/src/rules/no-require-imports.ts:49]: This is a copy-paste from https://github.com/typescript-eslint/typescript-eslint/blob/4a88de2c621ae645c070d995768006df8cb7e4df/packages/eslint-plugin/src/rules/no-shadow.ts#L329-L346

**Review by bradzacher — CHANGES_REQUESTED** [review]
thanks for doing this!
a few comments

**Comment by rafaelss95** [comment]
> instead of copy-pasting it - let's move this to an internal util within the util folder and share the code.

> https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/util

Do you want me to do this in this PR (and also use it in `no-shadow`)?

File: packages/eslint-plugin/src/rules/no-require-imports.ts [hunk 2]
```diff
@@ -18,13 +18,17 @@ export default util.createRule({
...
+      'CallExpression[callee.name="require"]'(
+        node: TSESTree.CallExpression,
       ): void {
-        context.report({
-          node: node.parent!,
-          messageId: 'noRequireImports',
-        });
+        const variable = getVariableByName(context.getScope(), 'require');
+
+        if (!variable?.identifiers.length) {
```
[the remark this answers is not in the document]
Comment (rafaelss95) [reply packages/eslint-plugin/src/rules/no-require-imports.ts:28]: `yarn check-spelling` complains about this:

Userland

[the remark this answers is not in the document]
Comment (bradzacher) [reply packages/eslint-plugin/src/rules/no-require-imports.ts:28]: feel free to hyphenate it - `user-land` it should be fine with

**Comment by bradzacher** [comment]
> (I can send another PR addressing this or maybe amend it to #3868). Let me know your thoughts.

Feel free to make another PR - keeps it clear and attributes each contribution to you :)

**Pull request merged**